src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -50,11 +50,19 @@ void shape_infer(const Squeeze* op,
         OPENVINO_SUPPRESS_DEPRECATED_END
 
         std::vector<int64_t> axes;
-        if (arg_rank.is_static() && axes_shape.is_static() && get_data_as_int64<T>(1, op, axes, constant_data)) {
-            OPENVINO_SUPPRESS_DEPRECATED_START
-            normalize_axes(op, arg_rank.get_length(), axes);
-            OPENVINO_SUPPRESS_DEPRECATED_END
-            unique_axes.reset(new std::set<int64_t>(axes.cbegin(), axes.cend()));
+        if (arg_rank.is_static() && axes_shape.is_static()) {
+            if (get_data_as_int64<T>(1, op, axes, constant_data)) {
+                // The values of `axes` input are known
+                OPENVINO_SUPPRESS_DEPRECATED_START
+                normalize_axes(op, arg_rank.get_length(), axes);
+                OPENVINO_SUPPRESS_DEPRECATED_END
+                unique_axes.reset(new std::set<int64_t>(axes.cbegin(), axes.cend()));
+            } else if (arg_rank.get_length() > 0 && shape_size(axes_shape.to_shape()) == 1) {
+                // The `axes` input must be a Parameter with single element to ensure uniqueness of axes
+                // only rank is deduced
+                output_shape = PartialShape::dynamic(arg_rank.get_length() - 1);
```

Comment:
If axes is parameter without any data for input parameter then you squeeze rank but what if input shape has no axis which is compatible with dimension `1`?

Reply:
The check for number of squeezable dimensions has been added. But previously output shape rank was dynamic in such case.

Reply:
Right, thanks, done:)